### PR TITLE
added CARNIVAL + differential gene expression and readded LDGM

### PR DESCRIPTION
**Pull request by MartinFXP**
title +
some minor additions/edits (e.g., help files, package suggests)

### CODE REVIEW

File: R/main.R [hunk 1]
```diff
@@ -15,7 +15,8 @@
 #' @param p_method character string. "mean", "sum" for standard summary
 #' functions, "hmp" for harmonic mean, "test" for the selfcontained
 #' test of package 'CombinePValue' or any method from package 'metap',
-#' e.g., "meanp" or "sump".
+#' e.g., "meanp" or "sump". Alternatively, "vcovHC" can improve results
```
Comment (kpj) [inline R/main.R:18]: What would you think of properly validating our parameters with something like `match.arg`?

File: inst/scripts/synthetic_benchmark/workflow/scripts/Carnival.R [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# G[lower.tri(G)] <- 0
+# diag(G) <- 0
+# rownames(G) <- colnames(G) <- 1:nrow(G)
+# 
+# X <- matrix(rnbinom(n = n*p, size = 100, mu = 1000),100,p)
+# Y <- matrix(rnbinom(n = n*p, size = 100, mu = 1000),100,p)
+# diff <- sample(1:10,2)
+# Y[,diff] <- rnbinom(n = 2*n, size = 100, mu = 100)
+# 
+# test <- carWrap(X,Y,G)
```
Comment (kpj) [inline inst/scripts/synthetic_benchmark/workflow/scripts/Carnival.R:48]: Is this commented out code needed or can it be removed?

File: inst/scripts/synthetic_benchmark/workflow/scripts/DGE.R [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# X <- matrix(rnbinom(n = n*p, size = 100, mu = 1000),100,p)
+# Y <- matrix(rnbinom(n = n*p, size = 100, mu = 1000),100,p)
+# diff <- sample(1:10,2)
+# Y[,diff] <- rnbinom(n = 2*n, size = 100, mu = 100)
+#
+# colnames(G) <- rownames(G) <- colnames(Y) <- colnames(X) <- 1:ncol(X)
+#
+# test <- dge_net(X,Y,G)
+#
+# test2 <- dce(G,X,Y,solver = "lm")
```
Comment (kpj) [inline inst/scripts/synthetic_benchmark/workflow/scripts/DGE.R:42]: Is this commented out needed?

File: inst/scripts/synthetic_benchmark/workflow/scripts/LDGM.R [hunk 4]
```diff
@@ -8,8 +8,11 @@ LDGM <- function(wt.X,mt.X) {
     write.table(wt.LC,file=wt.f,row.names=FALSE,col.names=FALSE,sep=",")
     write.table(mt.LC,file=mt.f,row.names=FALSE,col.names=FALSE,sep=",")
     theta.f <- paste0("Theta.",runif(1),".csv")
-    system(paste0("matlab -batch \"Sigma1 = readmatrix('",mt.f,"'); Sigma2 = readmatrix('",wt.f,"'); lambda = 0.26924; cd('LDGM'); Theta = differential_graph(Sigma1,Sigma2,lambda); writematrix(Theta,'",theta.f,"'); exit\""))
-    Theta <- as(read.csv(paste0("LDGM/",theta.f),header=FALSE),"matrix")
-    system(paste0("rm ", wt.f, " ", mt.f, " LDGM/", theta.f))
+    system(paste0("matlab -batch \"Sigma1 = readmatrix('",mt.f,"'); Sigma2 = readmatrix('",wt.f,"'); lambda = 0.26924; cd('workflow/scripts/LDGM'); Theta = differential_graph(Sigma1,Sigma2,lambda); writematrix(Theta,'",theta.f,"'); exit\""))
```
Comment (kpj) [inline inst/scripts/synthetic_benchmark/workflow/scripts/LDGM.R:11]: How is the value `0.26924` for lambda chosen?

**Review by kpj — CHANGES_REQUESTED** [review]
Looking snazzy!

File: inst/scripts/synthetic_benchmark/workflow/scripts/LDGM.R (hunk 4 above)
Comment (MartinFXP) [reply inst/scripts/synthetic_benchmark/workflow/scripts/LDGM.R:11]: From the package example. Added to the manuscript.

File: R/main.R (hunk 1 above)
Comment (MartinFXP) [reply R/main.R:18]: Probably a good idea.

File: inst/scripts/synthetic_benchmark/workflow/scripts/LDGM.R (hunk 4 above)
Comment (kpj) [reply inst/scripts/synthetic_benchmark/workflow/scripts/LDGM.R:11]: Makes sense!

File: R/main.R (hunk 1 above)
Comment (kpj) [reply R/main.R:18]: Great :D

File: R/main.R [hunk 5]
```diff
@@ -270,6 +272,10 @@ dce_nb <- function(
     conservative,
     log_level
 ) {
+    avail.tests <- c("lr","wald","vcovHC")
```
Comment (kpj) [inline R/main.R:275]: Please don't use periods in variable names.
Here you should use `avail_tests` instead.

More generally speaking, does it really improve readability to not simply define this list as an argument to `match.arg`?

**Review by kpj — CHANGES_REQUESTED** [review]

File: R/main.R [hunk 6]
```diff
@@ -272,10 +272,8 @@ dce_nb <- function(
     conservative,
     log_level
 ) {
-    avail.tests <- c("lr","wald","vcovHC")
-    test <- match.arg(test,avail.tests)
-    avail.p_methods <- c("hmp","meanp","sump","test")
-    p_method <- match.arg(p_method,avail.p_methods)
+    test <- match.arg(test, c("lr", "wald", "vcovHC"))
+    p_method <- match.arg(p_method, c("hmp", "meanp", "mean", "sum", "sump", "test"))
```
Comment (kpj) [inline R/main.R]:
```
p_method <- match.arg(
    p_method,
    c("hmp", "meanp", "mean", "sum", "sump", "test")
)
```
;)

**Review by kpj — CHANGES_REQUESTED** [review]

File: inst/scripts/synthetic_benchmark/workflow/scripts/DGE.R [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  design <- model.matrix(~group)
+  y <- estimateDisp(y,design)
+  fit <- glmQLFit(y,design)
+  qlf <- glmQLFTest(fit,coef=2)
+  fc <- qlf$table$logFC
+  pval <- qlf$table$PValue
+  pval[pval < 1e-50] <- 1e-50
+  edges <- which(G != 0, arr.ind = TRUE)
+  dce <- dcep <- G
+  for (i in 1:nrow(edges)) {
```
Comment (kpj) [inline inst/scripts/synthetic_benchmark/workflow/scripts/DGE.R:17]: `seq_len` should be used instead of `:`.

**Review by kpj — CHANGES_REQUESTED** [review]

**Comment by kpj** [comment]
Any idea why the test `(test-dce.R:219:3): different p-value computations work` is failing?
It's passing on master.

**Comment by MartinFXP** [comment]
> Any idea why the test `(test-dce.R:219:3): different p-value computations work` is failing?
> It's passing on master.

The only change within the main part should be in the function `topologically_ordering()`. I don't see atm how that can influence the test.

**Comment by kpj** [comment]
Shall I just merge this PR and see how it goes?

**Comment by MartinFXP** [comment]
Let's do one more round. Maybe it was just a glitch.

**Comment by kpj** [comment]
Seems like it didn't change 🤷

**Comment by MartinFXP** [comment]
I know whats going on. I'll have another go in a sec.

**Comment by kpj** [comment]
Nice! Good thing we have tests 😅
Should I merge it now?

**Comment by MartinFXP** [comment]
Sure.

**Pull request merged**